src/jsifier.js

```diff
@@ -423,12 +423,14 @@ function JSify(data, functionsOnly) {
 
       if (!SIDE_MODULE && !WASM_BACKEND) {
         if (USE_PTHREADS) {
-          print('var tempDoublePtr;');
-          print('if (!ENVIRONMENT_IS_PTHREAD) tempDoublePtr = ' + makeStaticAlloc(12) + ';');
+          print('// Pthreads fill their tempDoublePtr memory area into the pthread stack when the thread is run.')
+          // Main thread still statically allocate tempDoublePtr - although it could theorerically also use its stack
+          // (that might allow removing the whole tempDoublePtr variable altogether from the codebase? but would need
+          // more refactoring)
+          print('var tempDoublePtr = ENVIRONMENT_IS_PTHREAD ? 0 : ' + makeStaticAlloc(8) + ';');
```

Comment:
Should be be consistent and simply always put this pointer on the stack?

Reply:
Yeah, thinking the same, but did not want to change too much here for the CI, but first see if this gets through.